safe-global/web-core
# Refactor: Create Safe Status hooks

## What it solves

Resolves #932 

Redesign will be done in a separate PR

## How this PR fixes it

- Refactors the folder structure for create safe status related components and hooks
- Removes `AWAITING_WALLET` state and instead disables the retry button
- Simplifies `useSafeCreation` hook and adds more tests
- Moves `CreationStatus` component into the stepper
- Removes `usePendingSafe` hook
- Renames `useWatchSafeCreation` to `useSafeCreationEffects`

## ToDos

- [x] Implement test cases
- [x] Visually hide stepper when in status step

## How to test it

The following messages should be visible to the user during safe creation: https://www.notion.so/gnosis-safe/Revamp-bc8783e2eaf041d0b6a50c9a1641aee7

1. Create safe, confirm in wallet -> AWAITING, PROCESSING, SUCCESS
2. Create safe, reject in wallet -> AWAITING, WALLET_REJECTED
3. Create safe, confirm in wallet, reload the page -> PROCESSING, SUCCESS
4. Create safe, reject in wallet, reload the page -> AWAITING
5. Create safe, reject in wallet, disconnect wallet -> Retry button should be disabled and show a Tooltip on hover
6. Create safe, confirm in wallet, speed-up tx -> AWAITING, PROCESSING, SUCCESS
7. Create safe, confirm in wallet, cancel tx -> AWAITING, PROCESSING, ERROR
8. Create safe, confirm in wallet, reload page, speed-up tx -> PROCESSING, SUCCESS
9. Create safe, confirm in wallet, reload page, cancel tx -> PROCESSING, ERROR
10. Create safe, confirm in wallet, speed-up tx, reload page -> PROCESSING, SUCCESS
11. Create safe, confirm in wallet, cancel tx, reload page -> PROCESSING, ERROR
